store/sqlstore/file_info_store.go

```diff
@@ -76,17 +76,16 @@ func (fs SqlFileInfoStore) Get(id string) (*model.FileInfo, *model.AppError) {
 			Id = :Id
 			AND DeleteAt = 0`, map[string]interface{}{"Id": id}); err != nil {
 		if err == sql.ErrNoRows {
-			return nil, model.NewAppError("SqlFileInfoStore.Get", "store.sql_file_info.get.app_error", nil, "id="+id+", "+err.Error(), http.StatusNotFound)
+			return nil, store.NewErrNotFound("FileInfo", id)
 		}
-		return nil, model.NewAppError("SqlFileInfoStore.Get", "store.sql_file_info.get.app_error", nil, "id="+id+", "+err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrapf(err, "failed to get FileInfo with id=%s", id)
 	}
 	return info, nil
 }
 
-func (fs SqlFileInfoStore) GetWithOptions(page, perPage int, opt *model.GetFileInfosOptions) ([]*model.FileInfo, *model.AppError) {
+func (fs SqlFileInfoStore) GetWithOptions(page, perPage int, opt *model.GetFileInfosOptions) ([]*model.FileInfo, error) {
 	if perPage < 0 || page < 0 {
-		return nil, model.NewAppError("SqlFileInfoStore.GetWithOptions",
-			"store.sql_file_info.get_with_options.app_error", nil, fmt.Sprintf("page=%d and perPage=%d must be non-negative", page, perPage), http.StatusBadRequest)
+		return nil, store.NewErrInvalidInput("FileInfo", "<perPage, page>", fmt.Sprintf("<%d, %d>", perPage, page))
```

Comment:
Wondering if "ErrLimitExceeded" is a better choice here?

Reply:
sure!